services/Referral.js

```diff
@@ -7,7 +7,12 @@ class Referral {
    * @param {object} data
    */
   static async createReferral(object) {
-    return await ReferralModel.create(object);
+    try {
+      const newReferral = await ReferralModel.create(object);
+      return newReferral;
+    } catch (e) {
+      return null;
```

Comment:
Do we want to log a simple error message here so we know when this fails or is the plan to fail silently?

Reply:
I'm adding some logging for it.